application/client.core/src/app/environment/components/sidebar/shell/environment/component.ts

```diff
@@ -1,198 +1,125 @@
-import { Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
-import { IEnvironment, INewInformation } from '../input/component';
+import { Component, Input, OnDestroy } from '@angular/core';
+import { SidebarAppShellEnvironmentVariablesComponent } from './environment_variables/component';
+import { ShellService } from '../services/service';
+import { Subscription } from 'rxjs';
+import { Session } from '../../../../controller/session/session';
+import { MatSelectChange } from '@angular/material/select';
+
+import PopupsService from '../../../../services/standalone/service.popups';
+import TabsSessionsService from '../../../../services/service.sessions.tabs';
+import EventsSessionService from '../../../../services/standalone/service.events.session';
+
+import * as Toolkit from 'chipmunk.client.toolkit';
+
+interface IEditing {
+    variable: boolean;
+    value: boolean;
+}
+
+export interface IEnvironment {
+    variable: string;
+    value: string;
+    custom: boolean;
+    editing: IEditing;
+    selected: boolean;
+}
 
-import ContextMenuService, { IMenuItem } from '../../../../services/standalone/service.contextmenu';
+export interface INewInformation {
+    shell: string;
+    pwd: string;
+    env: IEnvironment[];
+}
 
 @Component({
     selector: 'app-sidebar-app-shell-environment',
     templateUrl: './template.html',
     styleUrls: ['./styles.less']
 })
 
-export class SidebarAppShellEnvironmentComponent implements OnInit, OnDestroy {
-
-    @Input() public information: INewInformation;
-    @Input() public setEnvironment: (information: INewInformation) => void;
-    @Input() public close: () => {};
-
-    @ViewChild('variableAdd') _ng_variableAdd: ElementRef<HTMLInputElement>;
+export class SidebarAppShellEnvironmentComponent implements OnDestroy {
 
-    public readonly _ng_variable: string = 'variable';
-    public readonly _ng_value: string = 'value';
-    public _ng_selected: IEnvironment;
-    public _ng_adding: boolean = false;
-    public _ng_height: number = 200;
-    public _ng_newVariable: string = '';
-    public _ng_newValue: string = '';
+    @Input() public service: ShellService;
 
-    private _prevValue: string = '';
-    private _prevVariable: string = '';
-    private _editing: IEnvironment | undefined;
+    public _ng_expanded: boolean = false;
 
-    constructor() { }
+    private _sessionID: string;
+    private _subscriptions: { [key: string]: Toolkit.Subscription | Subscription } = {};
+    private _logger: Toolkit.Logger = new Toolkit.Logger('SidebarAppShellEnvironmentComponent');
 
-    public ngOnInit() {
-        this._sortInformation();
-        this._ng_height = Math.round(window.innerHeight * .7);
+    constructor() {
+        this._sessionID = TabsSessionsService.getActive().getGuid();
```

Comment:
Please take into account
```
public getActive(): Session | undefined
```
`getActive()` can return `undefined`. In our lifecircle it isn't possbile situation, but as soon as some changes came, it can be a source of issues.

If we cannot be sure: does session exist or not, better to to move `this._sessionID = TabsSessionsService.getActive().getGuid();` into `ngInit` or `ngViewInit` and do related actions, like show message "session isn't available or something like it